src/Lib/Controller.php

```diff
@@ -204,6 +204,38 @@ protected function draw($view = null, $dir = null)
             ->fetch($view, $dir);
     }
 
+    /**
+     * Overwritable method called before the partial method
+     *
+     * @param string $view The name of the view file to render
+     * @param string $dir The directory to find the given view in
+     * @return array A list containing: (optional)
+     *
+     *  - view The name of the view file to render
+     *  - params An array of parameters to set in the view
+     *  - dir The directory to find the given view in
+     */
+    public function prePartial($view, $dir = null)
```

Comment:
`::partial` is in protected scope, but your `::prePartial`/`::postPartial` are public?

These methods also do not return an array as indicated in the comments? I'd expect it should return what it says it returns or otherwise be an abstract method in that case, but then the child classes would have to define them.

I'm a bit confused what the overall intent is here (out of the loop, I know), but I would expect these methods to take the same arguments as `::partial` (it's missing `$params`) since the comments indicate it is supposed to return params somehow.